test/test_inflate_small.cc

```
@@ -0,0 +1,98 @@
+                break;
+            EXPECT_EQ(err, Z_OK) << "row_size=" << row_size;
+
+            uint32_t processed = chunk - d_strm.avail_out;
+            out_pos += processed;
+            remaining -= processed;
+        }
+
+        PREFIX(inflateEnd)(&d_strm);
+
+        EXPECT_EQ(d_strm.total_out, data_size) << "row_size=" << row_size;
+        EXPECT_EQ(memcmp(decompressed, original, data_size), 0) << "row_size=" << row_size;
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)